lib/data_models/nordigen_transaction_model.dart

```diff
@@ -152,7 +154,7 @@ class TransactionData {
   final AmountData transactionAmount;
 
   /// Map of Report Exchange Rate.
-  final Map<String, dynamic>? currencyExchange;
+  final List<Map<String, dynamic>> currencyExchange;
```

Comment:
Shouldn't List<Map<String, dynamic>> be nullable?
It's better to include a transaction without currencyExchange to the test.
I don't posses all of transaction cases but can share the following:
```
{
                "transactionId": "uuid-1",
                "bookingDate": "2022-08-16",
                "valueDate": "2022-08-16",
                "transactionAmount": {
                    "amount": "-150.0",
                    "currency": "EUR"
                },
                "creditorName": "Someone",
                "creditorAccount": {
                    "iban": "DE87100110535353906"
                },
                "remittanceInformationUnstructured": "MoneyBeam",
                "remittanceInformationUnstructuredArray": [
                    "MoneyBeam"
                ],
                "bankTransactionCode": "ICDT"
            },
            {
                "transactionId": "uuid-2",
                "bookingDate": "2022-08-16",
                "valueDate": "2022-08-16",
                "transactionAmount": {
                    "amount": "300.0",
                    "currency": "EUR"
                },
                "debtorName": "JOHN DOE",
                "debtorAccount": {
                    "iban": "DE047008000543427500"
                },
                "bankTransactionCode": "PMNT-ESCT"
            },
            {
                "transactionId": "uuid-3",
                "bookingDate": "2022-08-15",
                "valueDate": "2022-08-15",
                "transactionAmount": {
                    "amount": "-3.1",
                    "currency": "EUR"
                },
                "creditorName": "TICKETSHOP",
                "remittanceInformationUnstructured": "-",
                "remittanceInformationUnstructuredArray": [
                    "-"
                ],
                "additionalInformation": "90ef-4147-aa2e",
                "bankTransactionCode": "PMNT-CCRD"
            }
```

Reply:
Have made the call to keep default currencyExchange as an Empty list if received null from Server. Also integrated your new mock data into Unit Tests. If everything seems alright, we can merge and publish.

Reply:
Well since it's a list data structure, having it empty if null made more sense to me than having it nullable. That way instead of always running null check operators on it (and also checking length to iterate over it), you can be guaranteed it's non null and just perform operations on it based on its `length` or `isEmpty` checks.

Definitely open to keeping it null though if there is a stronger point to keep it null.

Reply:
Just was thinking about it in general. For instance, consider the following 2 responses:
```
            {
                "transactionId": "uuid-3",
                "bookingDate": "2022-08-15"
            }
```
and 
```
            {
                "transactionId": "uuid-3",
                "bookingDate": "2022-08-15",
                "currencyExchange": []
            }
```
As you can see, they are different. However the package returns `currencyExchange ` as an empty array. In theory, there might be use-cases when such differentiation is required. Although, in practice maybe API never returns an empty array, but who knows.

Also, I notices that all other fields are nullable, even lists. I don't see any reason why `currencyExchange ` should be treated differently.

If you think I'm a paranoiac that also can be true  :). So, you can merge it. Anyways, in my use-case I don't need this field at all.

Reply:
Understood. Have made it nullable. Will merge in a few hours if there is no other issues.